expression/expressions/in.go

```diff
@@ -158,14 +172,18 @@ func (n *PatternIn) Eval(ctx context.Context, args map[interface{}]interface{})
 			return nil, err
 		}
 
-		if g, e := len(r.GetFields()), 1; g != e {
+		if g, e := len(r.GetFields()), columnNumber(n.Expr); g != e {
 			return false, errors.Errorf("IN (%s): mismatched field count, have %d, need %d", n.Sel, g, e)
 		}
 
-		res = make([]expression.Expression, 0)
+		res = make([]interface{}, 0)
 		// evaluate select and save its result for later in expression check
 		err = r.Do(ctx, func(id interface{}, data []interface{}) (more bool, err error) {
-			res = append(res, Value{data[0]})
+			if len(data) == 1 {
+				res = append(res, data[0])
+			} else {
+				res = append(res, data)
+			}
```

Comment:
not the same. if data has one column, the result is []interface{}, if more column, the result is [][]interface{}